src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableOrchestrationContext.cs

```diff
@@ -173,6 +177,111 @@ Task<TResult> IInterleavingContext.CallSubOrchestratorWithRetryAsync<TResult>(st
+            return ((IDurableOrchestrationContext)this).CallHttpAsync(req);
+        }
+
+        Task<DurableHttpResponse> IDurableOrchestrationContext.CallHttpAsync(HttpMethod method, Uri uri, ITokenSource tokenSource)
+        {
+            DurableHttpRequest req = new DurableHttpRequest(method, uri);
+            req.TokenSource = tokenSource;
+            return ((IDurableOrchestrationContext)this).CallHttpAsync(req);
+        }
+
+        async Task<DurableHttpResponse> IDurableOrchestrationContext.CallHttpAsync(DurableHttpRequest req)
+        {
+            DurableHttpResponse durableHttpResponse = await this.ScheduleDurableHttpActivityFunctionAsync(req);
+
+            HttpStatusCode currStatusCode = durableHttpResponse.StatusCode;
+
+            while (currStatusCode == HttpStatusCode.Accepted && this.Config.Options.HttpSettings.AsynchronousPatternEnabled)
+            {
+                Dictionary<string, StringValues> headersDictionary = new Dictionary<string, StringValues>(durableHttpResponse.Headers);
+                DateTime currUtcTime = this.InnerContext.CurrentUtcDateTime;
+                DateTime fireAt = default(DateTime);
+                if (headersDictionary.ContainsKey("Retry-After"))
+                {
+                    fireAt = this.InnerContext.CurrentUtcDateTime.AddSeconds(int.Parse(headersDictionary["Retry-After"]));
+                }
+                else
+                {
+                    fireAt = this.InnerContext.CurrentUtcDateTime.AddMilliseconds(this.Config.Options.HttpSettings.DefaultAsyncRequestSleepTime);
+                }
+
+                await this.InnerContext.CreateTimer(fireAt, CancellationToken.None);
+
+                DurableHttpRequest durableAsyncHttpRequest = this.CreateNewHttpRequestMessageForAsyncResponse(req);
+                durableAsyncHttpRequest = AddLocationHeaderToAsyncRequest(durableAsyncHttpRequest, durableHttpResponse);
+
+                durableHttpResponse = await this.ScheduleDurableHttpActivityFunctionAsync(durableAsyncHttpRequest);
+
+                currStatusCode = durableHttpResponse.StatusCode;
+            }
+
+            return durableHttpResponse;
+        }
+
+        private async Task<DurableHttpResponse> ScheduleDurableHttpActivityFunctionAsync(DurableHttpRequest req)
+        {
+            string serializedRequest = MessagePayloadDataConverter.HttpConverter.Serialize(req);
+            JToken durableHttpResponseJson = await this.CallDurableTaskFunctionAsync<JToken>(
+                functionName: HttpOptions.HttpTaskActivityReservedName,
+                functionType: FunctionType.Activity,
+                oneWay: false,
+                instanceId: null,
+                operation: null,
+                retryOptions: null,
+                input: serializedRequest);
+
+            JsonSerializer serializer = new JsonSerializer();
+            serializer.Converters.Add(new DurableHttpResponseJsonConverter(typeof(DurableHttpResponse)));
```

Comment:
Instead of manually creating `JsonSerializer` objects like this, can we use the [JsonConverterAttribute](https://www.newtonsoft.com/json/help/html/JsonConverterAttributeClass.htm) on the `DurableHttpResponse` class and the new `MessagePayloadDataConverter.HttpConverter` that you added?

Reply:
Adding the JsonConverterAttribute made it so serialization had to be specified for `DurableHttpResponse`, not only deserialization and that was complicating the serialization process. Connor and I discussed this and decided to make the `JsonSerializer `a static variable in the interest of time.

Reply:
Cool - I think this is the next best solution. Let's go with this.